pdns/recpacketcache.cc

```diff
@@ -66,6 +67,12 @@ bool RecursorPacketCache::checkResponseMatches(std::pair<packetCache_t::index<Ha
         i = i + labellen;
       }
 
+      if(iter->d_vstate == Bogus) {
+        if(t_bogusremotes)
+          t_bogusremotes->push_back(source);
+        if(t_bogusqueryring)
+          t_bogusqueryring->push_back(make_pair(qname, qtype));
+      }
```

Comment:
It feels a bit wrong to have this code here, IMHO the packet cache should not have any knowledge of the rings or the requestor's source address. Perhaps this logic could be moved in `doProcessUDPQuestion()`?